pkg/lock/simple_mem.go

```diff
@@ -0,0 +1,30 @@
+package lock
+
+import "sync"
+
+var locks map[string]*memLock
```

Comment:
To implement the same ErrorLogger interface, with Lock() possibly returning an error.